src/map.cpp

```diff
@@ -1892,19 +1891,14 @@ bool ServerMap::deleteBlock(v3s16 blockpos)
 			return false;
 		// It may not be safe to delete the block from memory at the moment
 		// (pointers to it could still be in use)
-		sector->detachBlock(block);
-		m_detached_blocks.push_back(block);
+		m_detached_blocks.push_back(sector->detachBlock(block));
 	}
 
 	return true;
 }
 
 void ServerMap::deleteDetachedBlocks()
 {
-	for (MapBlock *block : m_detached_blocks) {
-		assert(block->isOrphan());
-		delete block;
-	}
```

Comment:
I had not kept the assert because it caused unused variable warnings in release builds.

Added it back, with a void-cast.

<!--
```cpp
	for (const auto &block : m_detached_blocks) {
		assert(block->isOrphan());
		(void)block; // silence unused-variable warning in release builds
	}

	assert([&]{
		for (const auto &block : m_detached_blocks)
			if (!block->isOrphan())
				return false;
		return true;
	}());

	assert(std::all_of(m_detached_blocks.begin(), m_detached_blocks.end(),
			[](const auto &block) { return block->isOrphan(); }));
```
-->

Reply:
not sure if void cast forces iteration to happen but this should be fine